### PR TITLE
Fix updates to activation policy, add a test

### PR DESCRIPTION
**Pull request by trodge**
fixes https://github.com/hashicorp/terraform-provider-google/issues/13511

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
sql: fixed an issue with updating the `google_sql_database_instance.settings.activation_policy` field
```

### CODE REVIEW

File: mmv1/third_party/terraform/resources/resource_sql_database_instance.go.erb [hunk 1]
```diff
@@ -1501,11 +1501,33 @@ func resourceSqlDatabaseInstanceUpdate(d *schema.ResourceData, meta interface{})
...
+		if err != nil {
+			return err
+		}
+	}
+
 	// Check if the database version is being updated, because patching database version is an atomic operation and can not be
 	// performed with other fields, we first patch database version before updating the rest of the fields.
-	if v, ok := d.GetOk("database_version"); ok {
-		instance = &sqladmin.DatabaseInstance{DatabaseVersion: v.(string)}
+	if d.HasChange("database_version") {
```
Comment (melinath) [inline mmv1/third_party/terraform/resources/resource_sql_database_instance.go.erb:1529]: why is this change necessary?

File: mmv1/third_party/terraform/tests/resource_sql_database_instance_test.go.erb [hunk 2]
```diff
@@ -1512,6 +1512,32 @@ func TestAccSqlDatabaseInstance_rootPasswordShouldBeUpdatable(t *testing.T) {
 	})
 }
 
+func TestAccSqlDatabaseInstance_activationPolicy(t *testing.T) {
```
Comment (melinath) [inline mmv1/third_party/terraform/tests/resource_sql_database_instance_test.go.erb:1515]: It looks like this test could replace `TestAccSqlDatabaseInstance_mysqlMajorVersionUpgrade` - sql tests are long and expensive so this could be a good opportunity to consolidate, even if it's just two tests.

File: mmv1/third_party/terraform/tests/resource_sql_database_instance_test.go.erb [hunk 3]
```diff
@@ -1512,6 +1512,32 @@ func TestAccSqlDatabaseInstance_rootPasswordShouldBeUpdatable(t *testing.T) {
...
+	t.Parallel()
+
+	instanceName := "tf-test-" + randString(t, 10)
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccSqlDatabaseInstanceDestroyProducer(t),
+		Steps: []resource.TestStep{
```
Comment (melinath) [inline mmv1/third_party/terraform/tests/resource_sql_database_instance_test.go.erb:1524]: Need to have import steps here.

File: mmv1/third_party/terraform/tests/resource_sql_database_instance_test.go.erb [hunk 4]
```diff
@@ -2956,4 +2982,19 @@ resource "google_sql_database_instance" "main" {
...
\ No newline at end of file
+}
+
+func testGoogleSqlDatabaseInstance_activationPolicy(instance, databaseVersion, activationPolicy string) string {
+	return fmt.Sprintf(`
+resource "google_sql_database_instance" "instance" {
+  name                = "%s"
+  region              = "us-central1"
+  database_version    = "%s"
+  deletion_protection = false
```
Comment (melinath) [inline mmv1/third_party/terraform/tests/resource_sql_database_instance_test.go.erb:2993]: Since the field is updatable, it would be great to have deletion_protection be modifiable and set to `true` for most of the steps to ensure that the resource isn't accidentally deleted & re-created - then set it to `false` in a final step.

**Review by melinath — CHANGES_REQUESTED** [review]
overall looks reasonable, a couple questions & change requests below.

File: mmv1/third_party/terraform/resources/resource_sql_database_instance.go.erb (hunk 1 above)
Comment (trodge) [reply mmv1/third_party/terraform/resources/resource_sql_database_instance.go.erb:1529]: It prevents the provider from doing an unnecessary update request when the database version hasn't changed.

File: mmv1/third_party/terraform/tests/resource_sql_database_instance_test.go.erb (hunk 2 above)
Comment (trodge) [reply mmv1/third_party/terraform/tests/resource_sql_database_instance_test.go.erb:1481]: Done

File: mmv1/third_party/terraform/tests/resource_sql_database_instance_test.go.erb (hunk 3 above)
Comment (trodge) [reply mmv1/third_party/terraform/tests/resource_sql_database_instance_test.go.erb:1490]: Done

File: mmv1/third_party/terraform/tests/resource_sql_database_instance_test.go.erb (hunk 4 above)
Comment (trodge) [reply mmv1/third_party/terraform/tests/resource_sql_database_instance_test.go.erb]: Done

**Comment by melinath** [comment]
@trodge feel free to re-request review when this is ready.

**Review by melinath — APPROVED** [review]
LGTM
